src/ActiveLogin.Authentication.BankId.Api/BankIdSimulatedApiClient.cs

```diff
@@ -94,7 +95,23 @@ public async Task<AuthResponse> AuthAsync(AuthRequest request)
             return new AuthResponse(orderRef, autoStartToken);
         }
 
-        private string GetPersonalIdentityNumber(AuthRequest request)
+        public async Task<SignResponse> SignAsync(SignRequest request)
+        {
+            await SimulateResponseDelay().ConfigureAwait(false);
+
+            var personalIdentityNumber = GetPersonalIdentityNumber(request);
+            await EnsureNoExistingAuth(personalIdentityNumber).ConfigureAwait(false);
+
+            var orderRef = Guid.NewGuid().ToString();
+            var auth = new Auth(request.EndUserIp, orderRef, personalIdentityNumber);
+            _auths.Add(orderRef, auth);
+
+            var autoStartToken = Guid.NewGuid().ToString().Replace("-", string.Empty);
+
+            return new SignResponse(orderRef, autoStartToken);
+        }
+
+        private string GetPersonalIdentityNumber(IAuthRequest request)
```

Comment:
Take `string personalIdentityNumber` as input instead as I'm suggesting removing the `IAuthRequest `interface. See comment below.